hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaUtils.java

```diff
@@ -31,6 +31,31 @@ public class AvroSchemaUtils {
 
   private AvroSchemaUtils() {}
 
+  /**
+   * Generates fully-qualified name for the Avro's schema based on the Table's name
+   *
+   * NOTE: PLEASE READ CAREFULLY BEFORE CHANGING
+   *       This method should not change for compatibility reasons as older versions
+   *       of Avro might be comparing fully-qualified names rather than just the record
+   *       names
+   */
+  public static String getAvroRecordQualifiedName(String tableName) {
+    String sanitizedTableName = HoodieAvroUtils.sanitizeName(tableName);
+    return "hoodie." + sanitizedTableName + "." + sanitizedTableName + "_record";
+  }
+
+  // TODO java-doc, test
+  public static boolean isProjectionOf(Schema sourceSchema, Schema targetSchema) {
+    for (Schema.Field targetField : targetSchema.getFields()) {
+      Schema.Field sourceField = sourceSchema.getField(targetField.name());
```

Comment:
avro1.10    getField will throw exception instead of return null,
```
  public Field getField(String fieldname) {
    throw new AvroRuntimeException("Not a record: " + this);
  }
```

Reply:
how about nest field project, do we need support it.

Reply:
Not strictly required here, but a good catch! Will update to make sure it's handled as well